ctapipe/core/tests/test_tool.py

```diff
@@ -340,3 +342,30 @@ class MyTool(Tool):
         json.dump({"MyTool": {"foo": 5}}, f)
 
     assert run_tool(MyTool(), [f"--config={config}"]) == 2
+
+
+def test_tool_raises():
+    class ToolGood(Tool):
+        name = "Good"
+        description = "This runs without an exception."
+
+        def start(self):
+            print("All good.")
+
+    class ToolBad(Tool):
+        name = "Bad"
+        description = "This tool raises an exception."
+
+        def start(self):
+            raise ValueError("1 does not equal 0.")
+
+    assert run_tool(ToolGood(), raises=True) == 0
+
+    assert run_tool(ToolBad(), raises=False) == 1
+
+    with pytest.raises(ValueError):
+        ret = run_tool(ToolBad(), raises=True)
+
+    # `ret` does not exist, because `run_tool` raised an Exception
+    with pytest.raises(UnboundLocalError):
+        assert ret == 1
```

Comment:
`assert "ret" not in locals()`?

Reply:
Or just leave it out, there is no possibility of ret being set when the ValueError does not happen

Reply:
> there is no possibility of ret being set when the ValueError does happen

Yes, and I wanted to make that very very clear. But I can remove it

Reply:
I removed it